fakeredis/fakeredis.go

```diff
@@ -59,10 +62,33 @@ func (s *FakeRedisServer) Requests() []*resp.Msg {
 
 func (s *FakeRedisServer) Run(startedChan chan struct{}) {
 	var err error
+	var tcpListener *net.TCPListener
+	var unixListener *net.UnixListener
```

Comment:
Why use separate variables?  It looks like if you assign directly `s.listener = net.Listen*(...)` regardless of the actual `Listen*` function used, you could skip these, and also get rid of the switch in lines 100-107.